### PR TITLE
Feature/add nampi scripts

### PR DESCRIPTION
**Pull request by mspector**
let's talk about these changes to see if you like them before merging

### CODE REVIEW

File: nips2016_NAMPI/process/commentProcess.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          "groups": pcs,
+          "subject": "Private comment posted to a paper: \"" + note.content.title + "\"",
+          "message": "A submission to "+ conference+" has received a comment for the Program Chairs. \n\nTitle: "+note.content.title+"\n\nComment: "+note.content.comment+"\n\nTo view the comment, click here: "+baseUrl+"/forum?id=" + note.forum
+        };
+        return or3client.or3request( or3client.mailUrl, pc_mail, 'POST', token );
+      })
+    };
+
+    var getCommentEmails = function(replytoNoteSignatures){
+      console.log('replytoNoteSignatures before filter:',replytoNoteSignatures);
```
Comment (melisabok) [inline nips2016_NAMPI/process/commentProcess.js]: do we need all these console logs?

File: nips2016_NAMPI/process/commentProcess.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        promises.push(reviewerMailP);
+      };
+
+      if(visibleToPCs){
+        var pcMailP = getPCEmails();
+        promises.push(pcMailP);
+      }
+
+      var rootComment = note.forum == note.replyto;
+      var anonComment = replytoNoteSignatures.indexOf('(anonymous)')>-1 ? true : false;
```
Comment (melisabok) [inline nips2016_NAMPI/process/commentProcess.js]: replyNoteSignatures can be null, a null checking is needed

File: nips2016_NAMPI/process/commentProcess.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      var replytoNote = note.replyto ? result[1].notes[0] : null;
+      var replytoNoteSignatures = replytoNote ? replytoNote.signatures : null;
+
+      console.log('replytoNote',replytoNote);
+      console.log('replytoNoteSignatures',replytoNoteSignatures);
+
+      var promises = [];
+      console.log('note',note)
+      console.log('note.readers',note.readers)
+      var visibleToEveryone = note.readers.indexOf('everyone')>-1 ? true : false;
```
Comment (melisabok) [inline nips2016_NAMPI/process/commentProcess.js]: Could you use includes? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

```
var visibleToEveryone = note.readers.includes('everyone');
```

File: nips2016_NAMPI/process/commentProcess.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      console.log('replytoNoteSignatures after filter:',replytoNoteSignatures);
+      var comment_mail = {
+        "groups": replytoNoteSignatures,
+        "subject":"Your post has received a comment",
+        "message": "One of your posts has received a comment.\n\nTitle: "+note.content.title+"\n\nComment: "+note.content.comment+"\n\nTo view the comment, click here: "+baseUrl+"/forum?id=" + note.forum
+      };
+      return or3client.or3request( or3client.mailUrl, comment_mail, 'POST', token );
+    };
+
+    console.log('functions defined')
```
Comment (melisabok) [inline nips2016_NAMPI/process/commentProcess.js]: Remove logs or add semicolons at the end of the sentence

File: nips2016_NAMPI/process/officialReviewProcess.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    var or3client = lib.or3client;
+    
+    var origNote = or3client.or3request(or3client.notesUrl+'?id='+note.forum, {}, 'GET', token);
+    var list = note.invitation.replace(/_/g,' ').split('/');
+    list.splice(list.indexOf('-',1));
+    var conference = list.join(' ');
+
+
+    origNote.then(function(result){
+      var note_number = result.notes[0].number
```
Comment (melisabok) [inline nips2016_NAMPI/process/officialReviewProcess.js]: add ;

File: nips2016_NAMPI/process/officialReviewProcess.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+function(){
+    var or3client = lib.or3client;
+    
+    var origNote = or3client.or3request(or3client.notesUrl+'?id='+note.forum, {}, 'GET', token);
+    var list = note.invitation.replace(/_/g,' ').split('/');
```
Comment (melisabok) [inline nips2016_NAMPI/process/officialReviewProcess.js]: This piece of code is present in all the process functions, could be move it to the or3client? as a helper for the script developer.

File: nips2016_NAMPI/process/openReviewProcess.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    var or3client = lib.or3client;
+    
+    var origNote = or3client.or3request(or3client.notesUrl+'?id='+note.forum, {}, 'GET', token);
+    var list = note.invitation.replace(/_/g,' ').split('/');
+    list.splice(list.indexOf('-',1));
+    var conference = list.join(' ');
+
+
+    origNote.then(function(result){
+      var note_number = result.notes[0].number
```
Comment (melisabok) [inline nips2016_NAMPI/process/openReviewProcess.js]: add ;

File: nips2016_NAMPI/process/openReviewProcess.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      var author_mail = {
+        "groups": authors,
+        "subject": "Review of your submission to " + conference + ": \"" + note.content.title + "\"",
+        "message": "Your submission to "+ conference +" has received a public review.\n\nTitle: "+note.content.title+"\n\nReview: "+note.content.review+"\n\nTo view the review, click here: "+baseUrl+"/forum?id=" + note.forum
+      };
+
+      var promises = [
+        or3client.or3request( or3client.mailUrl, author_mail, 'POST', token )
+      ];
+      return Promise.all(promises)
```
Comment (melisabok) [inline nips2016_NAMPI/process/openReviewProcess.js]: add ;

**Comment by mspector** [comment]
it looks like using "use strict"; and naming the function name are both working fine

File: nips2016_NAMPI/process/responseInvitationProcess.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+function() {
+
+  var or3client = lib.or3client;
+  var hashKey = or3client.createHash(note.content.email, "4813408173804203984");
+  if(hashKey == note.content.key) {
+    if (note.content.response == 'Yes') {
+      console.log("Invitation replied Yes")
```
Comment (melisabok) [inline nips2016_NAMPI/process/responseInvitationProcess.js]: missing ;

File: nips2016_NAMPI/process/responseInvitationProcess.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      .then(function(result) {
+        var message = "The user " + note.content.email + " has accepted the invitation to serve as a reviewer."
+      })
+      .then(result => done())
+      .catch(error => console.log(error));
+    } else if (note.content.response == 'No'){
+      console.log("Invitation replied No")
+      //if a user is in the reviewers group, remove them from that group and add them to the reviewers-declined group
+      or3client.removeGroupMember(invitation.signatures[0]+'/reviewers', note.content.email, token)
+      or3client.addGroupMember(invitation.signatures[0] + '/reviewers-declined', note.content.email, token)
```
Comment (melisabok) [inline nips2016_NAMPI/process/responseInvitationProcess.js]: these two request calls are not linked,

**Pull request merged**